src/datasource/gateway.test.js

```diff
@@ -0,0 +1,310 @@
+  "http": {
+    "timeout": 0,
+    "headers": {
+      "Content-Type": "application/json",
+      "Blockchain-Subdomain": "mainnet",
+    }
+  },
+  "ppkFilePath": `${__dirname}/ppk.json`,
+  "passphrase": "L0c4ld3v3l0pment!",
+  "chainId": "mainnet",
+  "txFee": 10000,
+}
+
+describe("GatewayClient", () => {
+  const context = {
+    client: null,
+  };
+
+  const variables = {
+    height: 22788,
+    address: "04c56dfc51c3ec68d90a08a2efaa4b9d3db32b3b",
+    applictionAddress: "f0ff6a854af933e6a837b0bec885b0089313607f",
+    txHash: "E4A3EDE68171473996E68549C5CFC3C06B4865C35D194B8BDDA08908B4D394A6",
+    page: 1,
+    per_page: 1,
+    prove: false,
+    received: false,
+    blockchain: "",
+    stakingStatus: 2,
+    jailingStatus: 2,
+    addressHex: "",
+    encodedTxBytes: "",
+    txSendAmount: 1000,
+    recipientAddress: "0547fcf1f664df7e697db5a5514d3555c897d6f5",
+  }
+
+  const testCases = [
+    {
+      queryName: 'getBalance',
+      responseProperties: {
+        onSuccess: ['balance'],
+        onFailure: [''],
+      },
+      args: [variables.address, variables.height],
+    },
+    {
+      queryName: 'getTransaction',
+      responseProperties: {
+        onSuccess: [
+          'hash',
+          'height',
+          'index',
+          'proof',
+          'tx',
+          'tx_result',
+        ],
+        onFailure: [''],
+      },
+      args: [variables.txHash],
+    },
+    {
+      queryName: 'getApp',
+      responseProperties: {
+        onSuccess: [
+          'address',
+          'chains',
+          'jailed',
+          'max_relays',
+          'public_key',
+          'staked_tokens',
+          'status',
+          'unstaking_time',
+        ],
+        onFailure: [''],
+      },
+      args: [variables.applictionAddress, variables.height],
+    },
+    {
+      queryName: 'getNode',
+      responseProperties: {
+        onSuccess: [
+          'address',
+          'chains',
+          'jailed',
+          'public_key',
+          'service_url',
+          'status',
+          'tokens',
+          'unstaking_time',
+        ],
+        onFailure: [''],
+      },
+      args: [variables.address, variables.height],
+    },
+    {
+      queryName: 'getAccountTxs',
+      responseProperties: {
+        onSuccess: [
+          'total_count'
+          ['txs', 'tx_result', 'tx', 'stdTx', 'proof', 'hash', 'height', 'index']
+        ],
+        onFailure: [''],
+    },
+      args: [
+        variables.address,
+        variables.prove,
+        variables.received,
+        variables.page,
+        variables.per_page,
+      ],
+    },
+    {
+      queryName: 'sendRawTx',
+      responseProperties: {
+        onSuccess: [
+          'logs',
+          'txhash',
+        ],
+        onFailure: [''],
+      },
+      args: null, // if null, the getArgs will be used. For references with dynamic values
+      getArgs: () => [variables.addressHex, variables.encodedTxBytes]
+    },
+  ];
```

Comment:
This will allow for proper abstraction, to make tests configurable. Can help in CI too.